src/AutoWrapper.Test/Helper/CustomContractResolver.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace AutoWrapper.Test.Helper
+{
+    internal class CustomContractResolver<T> : DefaultContractResolver
+    {
+        public Dictionary<string, string> _propertyMappings { get; set; }
+        private readonly bool _useCamelCaseNaming;
+
+        public CustomContractResolver(bool useCamelCaseNaming)
+        {
+            _propertyMappings = new Dictionary<string, string>();
+            _useCamelCaseNaming = useCamelCaseNaming;
+            SetObjectMappings();
+        }
+
+        protected override string ResolvePropertyName(string propertyName)
+        {
+            var resolved = _propertyMappings.TryGetValue(propertyName, out string resolvedName);
+
+            if (_useCamelCaseNaming)
+                return (resolved) ? resolvedName.ToCamelCase() : base.ResolvePropertyName(propertyName.ToCamelCase());
+
+            return (resolved) ? resolvedName : base.ResolvePropertyName(propertyName);
+        }
+
+        protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
+        {
+            var prop = base.CreateProperty(member, memberSerialization);
+
+            return prop;
+        }
+
+        private void SetObjectMappings()
+        {
+            SetObjectMappings(typeof(T));
+        }
+
+        private void SetObjectMappings(Type classType)
+        {
+            foreach (PropertyInfo propertyInfo in classType.GetProperties(BindingFlags.DeclaredOnly | BindingFlags.Public | BindingFlags.Instance))
+            {
+
+                var wrapperProperty = propertyInfo.GetCustomAttribute<AutoWrapperPropertyMapAttribute>();
+                if (wrapperProperty != null)
+                {
+                    _propertyMappings.Add(wrapperProperty.PropertyName, propertyInfo.Name);
+                }
+
+                var type = propertyInfo.PropertyType;
+                if (type.IsClass)
+                    SetObjectMappings(propertyInfo.PropertyType);
+            }
+        }
+
+    }
+}
```

Comment:
@hueifeng Any particular reason why you are not using the Helper function already defined in AutoWrapper here? https://github.com/proudmonkey/AutoWrapper/blob/master/src/AutoWrapper/Helpers/CustomContractResolver.cs

Reply:
Since it has access level restrictions, I need to change to access level if I use it

Reply:
@hueifeng Oh I see. Just change them to `public` so you can reference them in the Test project. We would want to avoid code duplication as possible for the ease of maintainability.

Reply:
@proudmonkey OK, let me modify

Reply:
@hueifeng Looks good to me now. Should I merge?

Reply:
@proudmonkey Ok,merge